jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java

```diff
@@ -286,6 +286,31 @@ public static void putIntLittleEndian(ByteBuffer buffer, int value)
         flipToFlush(buffer, p);
     }
 
+    /**
+     * Slice a buffer given an offset and a length.
+     * @param buffer the buffer to slice
+     * @param offset the offset
```

Comment:
This should really be called "numBytesToSkip" or something.

You mentioned earlier that this is not an absolute index/position but the number of bytes _from the current position_. I can't remember why this is necessary but it is different from `ByteBuffer#slice`, and there's a non-negligible risk that this will bite us down the road.

Even better, if we could just make that an absolute offset (and perhaps call buffer.position() on the outside where necessary), we'd completely avoid this subtle semantic mismatch

Reply:
Whilst I agree that an `index` might more in tune with the existing `ByteBuffer.slice(int, int)`, I don't think `offset` is that confusing as it is used in the JVM `write` signatures and in our Content API.    If this is naturally used with an offset in the calling locations, then offset if fine.

Reply:
Yeah, sadly Java and some RFC don't always agree. Fingers crossed that this won't hurt in the future.

Can we please add this explanation somehow in the Javadocs?

Reply:
> @gregw The problem I see is the word "offset" being ambiguous. It's not the absolute offset from the beginning (as in `new String(char[] value, int offset, int count)`, for example).

@kohlschuetter it is AN offset, but that can mean different things.   A name cannot completely convey the semantics, as even "index" and "position" can be offsets, as `arrayOffset` of a `ByteBuffer` maybe non zero.
I think offset to the current contents is a reasonable interpretation, but let's make sure the javadoc makes it clear.